hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -450,6 +451,10 @@ runTest :: Test a -> (Either Failure a, [Log])
 runTest =
   runIdentity . runTestT
 
+mapTestT :: (forall b. m b -> n b) -> TestT m a -> TestT n a
```

Comment:
FWIW this might be slightly more powerful/useful if it was:

```hs
mapTestT :: Functor m => (m a -> n b) -> TestT m a -> TestT n b
mapTestT f =
  mkTestT . first (fmap f) . runTestT
```

Reply:
Sorry, maybe something more like:

```hs
mapTestT :: Applicative m => (m a -> n b) -> TestT m a -> TestT n b
mapTestT f =
  mkTestT . (=<<) (\(e, l) -> fmap (flip (,) l) $ traverse f e) . runTestT
```

Reply:
The more general version seems better.

Reply:
Yeah without the more general version it's just the same as `hoist`